zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala

```diff
@@ -6,12 +6,35 @@ import org.specs2.mutable.Specification
 class JSONParserTest extends Specification with TestQuery {
 
   val parser = new JSONParser
+
   "JSONParser" should {
-    "parse the by (spatial, time) count" in {
-      //val actualQuery = parser.parse(stateHourCountJSON)
-      //val expectQuery = new Query(schema.dataset, Seq.empty, Seq.empty, Seq.empty,
-      //Some(GroupStatement(Seq(byState, byHour), Seq(aggrCount))), None)
-      //actualQuery must_== expectQuery
+    "parse the hourly count request" in {
+      val actualQuery = parser.parse(hourCountJSON)
+      val expectQuery = Query(schema.dataset, Seq.empty, Seq.empty, Seq.empty,
+                              Some(GroupStatement(Seq(byHour), Seq(aggrCount))), None)
+      actualQuery.groups must_== expectQuery.groups
```

Comment:
Why did you verify the groups match in addition to verify if the whole query matches ?